### ISSUE TITLE
Memory leak when unauthorized

### ISSUE THREAD

**Issue by maxramqvist** [marcinbudny/eventstore_exporter#17]
Hi,
Great service, great dashboard! :)
Found a (must be?) memory leak. If the eventstore_exporter gets bad credentials - memory usage just keeps growing. Up to several gigabytes at least. Seems to be around 800mb / day for us.

When the exporter has the correct credentials memory usage is still high - but doesn't seem to grow totally without bounds.
The memory usage of eventstore_exporter where credentials are correct (newly restarted process):
```bash
pmap -d $(pgrep eventstore_expo) | tail -n1 | awk '{print $4}'
139720K
```
eventstore_exporter without creds, running for 4 hours:
```bash
pmap -d $(pgrep eventstore_ex) | tail -n1 | awk '{print $4}'
238356K
```

**Comment by marcinbudny** [comment]
Hi,
Thanks for pointing this out. I will take a look.

In the meantime, can you please open the `/metrics` endpoint on the exporter in a web browser and check if you see anything suspicious, like e.g. large cardinality of the metrics?

Also please let me know what is your exact setup: version of the ESDB and settings you use on the exporter.

**Comment by marcinbudny** [comment]
The problem should now be fixed in 0.10.1. Let me know.

As for high memory consumption in normal conditions, I guess it is a property of an application written in Go. I get 100MB from pmap when inspecting an application that just sleeps in a loop.

**Comment by marcinbudny** [comment]
Closing due to lack of activity
